Added grav-admin blueprints into clean branch

aashishvasu opened the pull request:
Hi, this is the clean branch with just the blueprint files in the PR. Do have a look and tell me if this is alright. I shall follow a similar method if I have something to contribute to the master theme.

jasonccox commented:
It looks like theses can be structured a bit more simply. Take a look at the second example [here](https://learn.getgrav.org/16/forms/blueprints/example-page-blueprint) to see what I mean.

jasonccox commented on blueprints/blog-post.yaml:
```diff
@@ -0,0 +1,22 @@
+          type: tab
+          title: Blog Post
+          fields:
+            header.image:
```
I'd like this to contain an explanation that this image will be displayed next to the post title/summary in the list of blog posts.

aashishvasu replied:
I dont think I can add a description at the title level. And I didnt want to crowd the description for the image. I could add a hover help tag for it

jasonccox replied:
Okay, let's do the help tag.

jasonccox commented on blueprints/blog-post.yaml:
```diff
@@ -0,0 +1,22 @@
+                  preview_images: true
+                  description: Choose from an image uploaded to the Content section of this page.
+                  accept:
+                    - .jpg
```
Let's also add `.jpeg` and `.gif`

jasonccox commented on blueprints/blog-post.yaml:
```diff
@@ -0,0 +1,22 @@
+                  label: Summary Image
+                  folder: 'self@'
+                  preview_images: true
+                  description: Choose from an image uploaded to the Content section of this page.
```
```suggestion
                  description: Choose from an image uploaded to the Page Media section of the Content tab.
```

aashishvasu commented:
> 
> 
> It looks like theses can be structured a bit more simply. Take a look at the second example [here](https://learn.getgrav.org/16/forms/blueprints/example-page-blueprint) to see what I mean.

I read the link you shared. I'm not exactly sure which part you're referring to. Are you talking about whitespacing it to make it more readable? Or adding the implementation to the Advanced (or one of the other pre-existing) tabs?

Edit: If you meant moving the custom functionality to the Content tab, I'm all for it. I tested it just now, and it does look/feel better.

jasonccox commented:
I did not mean moving the functionality to the Content tab, but if you think it works better that way then I say go for it. I'll go ahead and finish reviewing the rest of this now and wait for any further comments on how the files are structured until you make that change.

jasonccox commented on blueprints/modular.yaml:
```diff
@@ -0,0 +1,31 @@
+              type: toggle
+              label: Submenu for modules
+              highlight: 1
+              description: If this is activated, each child module will be a navigable submenu item under the parent page in the navbar.
```
```suggestion
              description: If activated, each child module will be a navigable sub-menu item under the parent page in the navigation bar.
```

jasonccox commented on blueprints/modular.yaml:
```diff
@@ -0,0 +1,31 @@
+          fields:
+            header.modules_in_menu:
+              type: toggle
+              label: Submenu for modules
```
```suggestion
              label: Show Modules in Navigation Menu
```

jasonccox commented on blueprints/modular.yaml:
```diff
@@ -0,0 +1,31 @@
+                1: Yes
+                0: No
+              validate:
+                type: bool
```
Let's put new lines between the fields just to make this easier to read.

aashishvasu replied:
Agreed. I havent really paid attention to the formatting so far. I will go over all the docs and do a pass of formatting the files.

jasonccox commented on blueprints/modular.yaml:
```diff
@@ -0,0 +1,31 @@
+                type: bool
+            header.single_page_menu:
+              type: toggle
+              label: Display at root level
```
```suggestion
              label: Show Single-Page Menu
```

jasonccox commented on blueprints/modular.yaml:
```diff
@@ -0,0 +1,31 @@
+              type: toggle
+              label: Display at root level
+              highlight: 1
+              description: If this is activated, each child module will be a navigable menu item at root menu level (useful for single page websites).
```
```suggestion
              description: If activated, each child module will be a top-level navigable menu item, and no other links will be shown in the navigation menu.
```

jasonccox commented on blueprints/modular/about.yaml:
```diff
@@ -0,0 +1,30 @@
+                    type: textarea
+                .icon:
+                  type: iconpicker
+                  label: FA Icon
```
```suggestion
                  label: FontAwesome Icon Classes
```

jasonccox commented on blueprints/modular/about.yaml:
```diff
@@ -0,0 +1,30 @@
+            header.details:
+              name: details
+              type: list
+              label: Details
```
Let's add a description of this field.
```suggestion
              label: Details
              description: Sub-sections to be shown after the page's Markdown content. Three details typically looks best.
```

aashishvasu replied:
Yeah I should have definitely put a description in here.

jasonccox commented on blueprints/modular/about.yaml:
```diff
@@ -0,0 +1,30 @@
+                  validate:
+                    type: textarea
+                .icon:
+                  type: iconpicker
```
This isn't showing previews for me, so it might be better to just use a text field.

aashishvasu replied:
I'm trying to figure out a way to have them show up, I think its because grav is loading some older version of FA for itself. Will change it to iconpicker again after I figure it out

jasonccox commented on blueprints/modular/about.yaml:
```diff
@@ -0,0 +1,30 @@
+                .title:
+                  type: text
+                  label: Title
+                  validate:
```
I don't think we need this.

jasonccox commented on blueprints/modular/about.yaml:
```diff
@@ -0,0 +1,30 @@
+                .content:
+                  type: text
+                  label: Content
+                  validate:
```
Don't need this one either.

jasonccox commented on blueprints/modular/about.yaml:
```diff
@@ -0,0 +1,30 @@
+                  label: FA Icon
+                  description: For example, `fas fa-envelope`.
+                .content:
+                  type: text
```
```suggestion
                  type: textarea
```

jasonccox commented on blueprints/modular/contact.yaml:
```diff
@@ -0,0 +1,39 @@
+              type: toggle
+              label: Show Contact Form
+              highlight: 1
+              help: This will select whether to show the form defined in the form par
```
```suggestion
              description: The form must be defined on the parent modular page.
```

jasonccox commented on blueprints/modular/contact.yaml:
```diff
@@ -0,0 +1,39 @@
+                0: No
+              validate:
+                type: bool
+
```
```suggestion
```

jasonccox commented on blueprints/modular/contact.yaml:
```diff
@@ -0,0 +1,39 @@
+              fields:
+                header.button.link:
+                  type: text
+                  label: Mailto Link
```
I want this to be generic so someone could link to Twitter, FB, or whatever else they want.
```suggestion
                  label: Link
```

aashishvasu replied:
Ah right, sure

jasonccox commented on blueprints/modular/contact.yaml:
```diff
@@ -0,0 +1,39 @@
+                header.button.link:
+                  type: text
+                  label: Mailto Link
+                  description: For Example 'mailto:youremail@here.com'
```
```suggestion
                  description: For example, 'mailto:youremail@here.com'.
```

jasonccox commented on blueprints/modular/contact.yaml:
```diff
@@ -0,0 +1,39 @@
+                  label: Button Text
+                header.button.icon:
+                  type: iconpicker
+                  label: FA Icon
```
```suggestion
                  label: FontAwesome Icon Classes
```

jasonccox commented on blueprints/modular/contact.yaml:
```diff
@@ -0,0 +1,39 @@
+                  type: text
+                  label: Button Text
+                header.button.icon:
+                  type: iconpicker
```
Let's use a `text` field here as well.

jasonccox commented on blueprints/modular/intro.yaml:
```diff
@@ -0,0 +1,36 @@
+              type: toggle
+              label: Use Footer Links
+              highlight: 1
+              help: This will let the section use the links defined in the footer section
```
```suggestion
              help: If activated, the Footer Links defined in the theme configuration will be shown with the links defined here.
```

jasonccox commented on blueprints/modular/intro.yaml:
```diff
@@ -0,0 +1,36 @@
+        buttons:
+          type: tab
+          title: Intro
+          fields:
```
Any reason why you didn't add `header.links`?

aashishvasu replied:
Must have slipped my mind. I have already implemented it for myself, I think its not a part of this branch. Will add.

jasonccox commented on blueprints/modular/intro.yaml:
```diff
@@ -0,0 +1,36 @@
+                0: No
+              validate:
+                type: bool
+            header.image:
```
Let's just let them use the Page Media section of the Content tab and have this be a description of that. For example,
```yaml
image: 
  type: display
  size: large
  label: Choosing a Custom Image
  content: Upload an image to the Page Media section of the Content tab to set the Intro image.
```

aashishvasu replied:
Yeah it is definitely more consistent with how I implemented the filepicker in other blueprints. Will change.

jasonccox commented on blueprints/modular/intro.yaml:
```diff
@@ -0,0 +1,36 @@
+                - image/*
+            header.img_alt:
+              type: text
+              label: Image Caption
```
This could be deceptive because alt text only appears if the image can't load.
```suggestion
              label: Image Alt Text
```

jasonccox commented on blueprints/modular/intro.yaml:
```diff
@@ -0,0 +1,36 @@
+            header.img_alt:
+              type: text
+              label: Image Caption
+              validate:
```
Don't need this.

jasonccox commented on blueprints/modular/projects.yaml:
```diff
@@ -0,0 +1,42 @@
+      fields:
+        features:
+          type: tab
+          title: Projects
```
```suggestion
          title: Projects
          description: Project tiles to be shown after the page's Markdown content.
```

jasonccox commented on blueprints/modular/projects.yaml:
```diff
@@ -0,0 +1,42 @@
+                .title:
+                  type: text
+                  label: Title
+                  validate:
```
Don't need this.

jasonccox replied:
We can delete this `validate` attribute on text fields. There are several other instances of this that I'll go ahead and mark.

jasonccox commented on blueprints/modular/projects.yaml:
```diff
@@ -0,0 +1,42 @@
+                  label: Image
+                  folder: 'self@'
+                  preview_images: true
+                  description: Choose from an image uploaded to the Content section of this page.
```
```suggestion
                  description: Choose from an image uploaded to the Page Media section of the Content tab.
```

jasonccox commented on blueprints/modular/projects.yaml:
```diff
@@ -0,0 +1,42 @@
+                  folder: 'self@'
+                  preview_images: true
+                  description: Choose from an image uploaded to the Content section of this page.
+                  accept:
```
Let's also allow `.jpeg` and `.gif` here.

jasonccox commented on blueprints/modular/projects.yaml:
```diff
@@ -0,0 +1,42 @@
+                .description:
+                  type: text
+                  label: Description
+                  description: Describe the project in a few words
```
```suggestion
```

jasonccox commented on blueprints/modular/projects.yaml:
```diff
@@ -0,0 +1,42 @@
+                  type: text
+                  label: Description
+                  description: Describe the project in a few words
+                  validate:
```
Once again, not needed.

jasonccox commented on blueprints/modular/projects.yaml:
```diff
@@ -0,0 +1,42 @@
+                    type: textarea
+                .link:
+                  type: text
+                  label: Link to project
```
```suggestion
                  label: Link to Project
```

jasonccox commented on blueprints/modular/projects.yaml:
```diff
@@ -0,0 +1,42 @@
+            header.projects:
+              name: projects
+              type: list
+              label: Projects
```
```suggestion
              label: Projects
              description: Project tiles to be shown after the page's Markdown content.
```

aashishvasu replied:
I originally had a description on this, but for some reason descriptions dont show up on lists. Can you confirm?

jasonccox replied:
You're right. A help tag does show up though - can you do that instead?

jasonccox commented:
Alright, I've made an initial pass at the whole thing. Take a look at my comments and feel free to respond if you have questions or disagree with anything I've said. I'll take a final look through everything once you're done. Thanks again for contributing!

aashishvasu commented:
I've made the changes you suggested, and also added a decent amount of whitespacing for legibility.

Additionally, about adding the blueprint stuff directly to the content bar, I have implemented that in the about.yaml file (commit bddced8). Let me know if it works for you, else I can revert that bit and keep all of them to be their separate tabs.

jasonccox commented:
Hi Aashish, I just wanted to let you know that I am still aware of your PRs. I've been busy with other things for the last few days, but I hope to spend some more time on these this weekend or early next week. Sorry for the delay!

aashishvasu commented:
Hey Jason, No problem at all man. Since I did the PR's I got pretty busy myself. Take your time.

jasonccox commented:
I definitely like moving things to the Content tab. Let's go ahead and do that with all of the blueprints.

jasonccox commented on blueprints/blog-post.yaml:
```diff
@@ -0,0 +1,25 @@
+                  label: Summary Image
+                  folder: 'self@'
+                  preview_images: true
+                  description: Choose from an image uploaded to the Page Media section of the Content tab.
```
Once this gets moved into the Content tab we can go ahead and change the description.
```suggestion
                  description: Choose from an image uploaded to the Page Media section.
```

jasonccox commented on blueprints/modular/intro.yaml:
```diff
@@ -0,0 +1,58 @@
+              label: Custom Intro Image
+              folder: 'self@'
+              preview_images: true
+              description: Choose from an image uploaded to the Page Media section of the Content tab to set the Intro image.
```
We can change this one as well since it'll be in the Content tab.
```suggestion
              description: Choose from an image uploaded to the Page Media section to set the Intro image.
```

jasonccox commented on blueprints/modular/projects.yaml:
```diff
@@ -0,0 +1,46 @@
+                .title:
+                  type: text
+                  label: Title
+                  validate:
```
```suggestion
```

jasonccox commented:
This is looking great! Once you make those last few changes I'll take a final look and then merge it in!

aashishvasu commented:
I've made all the changes to the blueprints, including shifting descriptions to help tags where necessary

jasonccox commented:
Alright, I'm going to go ahead and merge this! Great work.

I'd like to create a `Contributors` section in the readme and add you to it as well - is there anything you'd like me to link people to (website, email, social media, etc.)?

aashishvasu commented:
Sweet! I saw you linked to my github already, thats fine with me. I would have suggested my website, but seeing as I'm using this theme to update it right now, github works better haha
